### PR TITLE
MAINT: Setup poetry

### PR DESCRIPTION
**Pull request by kaitj**
## Proposed changes

This PR starts to integrate `poetry` for package management, including updating the `README`, and the actions workflow. AFAIK it won't automatically handle the packages and versions, but it is cleaner to use. The pre-release version (1.2.0a2) introduced dependency groups (eg. dev dependencies, actions dependencies, etc). that get installed alongside the default dependency. A `poetry.lock` file is included as any installs will always attempt to match the versions found in this file first, even if another version is specified in the `pyproject.toml`. Idea behind `poetry` is to have anyone contributing to the repo to be in the same environment (as much as possible).

With this, it will also automatically setup the precommit hook if `git` commands are run from within the virtual environment created by `poetry` (via `poetry shell`).

I have left the `setup.py` and `requirements.txt` in the repo for now, but should set a date for deprecation / removal if switch to `poetry` is made. `poetry` will generate a `requirements.txt` file with a simple command, though they recommend against going this route.

## Types of changes

What types of changes does your code introduce? Put an `x` in the boxes that apply

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionalitiy)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Other (if none of the other choices apply)

## Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you are unsure about any of the choices, don't hesitate to ask!

- [ ] Changes have been tested to ensure that fix is effective or that a feature works.
- [ ] Changes pass the unit tests
- [ ] Code has been run through [`black`](https://github.com/psf/black) with the `-l 79` flag.
- [x] I have included necessary documentation or comments (as necessary)
- [ ] Any dependent changes have been merged and published

## Notes
All PRs will undergo the unit testing before being reviewed. You may be requested to explain or make additional changes before the PR is accepted.

_PR template was adopted from [appium](https://github.com/appium/appium/blob/master/.github/PULL_REQUEST_TEMPLATE.md)_

### CODE REVIEW

**Comment by kaitj** [comment]
Wait until transition from heroku as the `requirements.txt` is required.

**Comment by kaitj** [comment]
I started to go through this action item for the migration.

* Some modules for quality purposes have been added to the `pyproject.toml`, but have not yet been setup.
* I was testing the `poetry build` command, and one thing we'll have to decide on is "afids-valdiator" vs "afidsvalidator" unless there is a way around that. I've changed the name in the `pyproject.toml` file for now to be "afidsvalidator" but this is an easy fix. The other way was to go through and update `manage.py` but wasn't sure if that would break anything.
* Took advantage of this PR to also rename "validator.html" -> "app.html" for URL purposes, and this has also been updated in the navbar.

**Comment by tkkuehn** [comment]
I'm just as happy calling it afidsvalidator if that's easier!

**Comment by kaitj** [comment]
This should be good for a first look. The CI workflow is working now, but failing because its not passing the quality checks.

File: .github/workflows/afids-validator_ci.yml [hunk 1]
```diff
@@ -86,18 +100,35 @@ jobs:
...
-      - name: Install black
-        run: |
-          python -m pip install --upgrade pip
-          pip install black
+      - name: Install poetry
+        uses: snok/install-poetry@v1
+        with:
+          version: 1.2.0
+          virtualenvs-create: true
+          virtualenvs-in-project: ture
```
Comment (tkkuehn) [inline .github/workflows/afids-validator_ci.yml:108]: Typo: `ture` -> `true`

**Comment by kaitj** [comment]
Huh...some of the things you caught I thought I had fixed. Maybe I didn't push them.

File: README.md [hunk 2]
```diff
@@ -9,28 +9,37 @@
...
-* heroku
 
 ### Setup for local testing
-1. Git clone the afids-validator repository `git clone https://github.com/afids/afids-validator.git`
-3. Set up python virtual environment `python -m virtualenv <venv directory>`
-4. In virtual environment, install required modules `pip install -r requirements.txt --no-cache-dir`
-5. Create a superuser via postgres `sudo createuser --interactive`
-6. Create a database via postgres `createdb fid_db`
-7. Set password for the created database
+1. Git clone the fid-validator repository `git clone https://github.com/afids/afids-validator.git`
```
[the remark this answers is not in the document]
Comment (kaitj) [reply README.md:30]: Must have been from a rebase, fixed in latest commit

**Comment by kaitj** [comment]
If this look good to you @tkkuehn, it can be merged. I'll try to go through and create a quality task similar to snakebids / pre-commit on a new branch so we can get past the linting not passing.

**Comment by tkkuehn** [comment]
Do you know what's going on with the "Lint code" job not running for 3.8 and 3.9? Otherwise this looks to be in good shape.

**Comment by kaitj** [comment]
> Do you know what's going on with the "Lint code" job not running for 3.8 and 3.9? Otherwise this looks to be in good shape.

I assumed it stopped because 3.10 failed, which would lead to the action failing all together.

**Pull request merged**